src/fileparse.py

```diff
@@ -0,0 +1,103 @@
+import csv
+import logging
+from pathlib import Path
+from typing import Any, Callable, List, TextIO, Union, Dict
+
+
+def parse_csv(
+    file_path_or_buffer: Union[str, Path, TextIO],
+    select: List[str] = None,
+    convert_fn: Union[List[Callable], Dict[str, Callable]] = None,
+    has_headers: bool = True,
+    delimiter: str = ",",
+    verbose: bool = True,
+) -> Union[List[tuple], List[Dict[str, Any]]]:
+    """
+    Parse a csv file into a list.
+
+    :param file_path_or_buffer: Path to the csv file or file-like object
+    :param select: Columns to select
+    :param convert_fn: List or dictionary of functions to apply to each column
+    :param has_headers: Indicator if file has headers.
+                        If False, creates a list of tuples otherwise a list of dictionaries
+    :param delimiter: Type of delimiter. Defaults to ','
+    :param verbose: Indicates if errors should be logged
+    :raises RuntimeError: Raises RuntimeError if arguments are not compatible with each other.
+                          If select is given then has_headers must be True.
+                          Keys in convert_fn must be the same values in select.
+                          If there are headers then convert_fn must be a dict of col_name:type.
+                          Keys in convert_fn must match the name of the headers.
+    :return: List of items in the file
+    """
+
+    if not has_headers and select:
+        raise RuntimeError("'select' argument requires column headers")
+    if (
+        select
+        and convert_fn
+        and isinstance(convert_fn, dict)
+        and set(select) != set(convert_fn.keys())
```

Comment:
No need to call the `.keys()` when you apply the `set` it will convert the keys of the dictionary automatically:
```suggestion
        and set(select) != set(convert_fn)
```

Reply:
Another option is to use `.keys()` without `set` since it is redundant. The `.keys` will return a dict_keys which support set operations. The following will work:
```python
set(select) != convert_fn.keys()
```

Reply:
Changed to `set(select) != convert_fn.keys()`.